File: benchmarks/micro/DeadStackFieldStore.c
#include <stdint.h>
#include <unistd.h>

typedef struct Pair {
    int64_t left;
    int64_t right;
} Pair;

static int64_t accumulate(int64_t seed, int64_t salt) {
    Pair pair = { seed, salt };
    pair.left = seed + salt;
    pair.right = seed ^ salt;
    return ((seed * 3) + salt) % 1000000007;
}

int main(void) {
    int64_t total = (int64_t)getpid();

    for (int32_t i = 0; i < 250000; i += 1) {
        total += accumulate(total, (int64_t)i);
        total %= 1000000007;
    }

    return total <= 0 ? 1 : 0;
}

File: benchmarks/micro/DeadStackFieldStore.rs
#![allow(unused_assignments, unused_variables)]

struct Pair {
    left: i64,
    right: i64,
}

fn accumulate(seed: i64, salt: i64) -> i64 {
    let mut pair = Pair {
        left: seed,
        right: salt,
    };
    pair.left = seed + salt;
    pair.right = seed ^ salt;
    ((seed * 3) + salt) % 1_000_000_007
}

fn main() {
    let mut total = std::process::id() as i64;

    for i in 0_i32..250_000 {
        total += accumulate(total, i as i64);
        total %= 1_000_000_007;
    }

    if total <= 0 {
        std::process::exit(1);
    }
}
